src/parse_coin_config.c

```diff
@@ -15,10 +15,13 @@ void set_ledger_application_name(buf_t *application_name) {
     for (size_t i = 0; i < sizeof(appnames_aliases) / sizeof(appnames_aliases[0]); i++) {
         if (!strcmp((const char *) application_name->bytes,
                     (char *) (PIC(appnames_aliases[i].foreign_name)))) {
+            const size_t app_name_size = strlen((char *) PIC(appnames_aliases[i].app_name));
             // Copy the correct application name.
-            strcpy((char *) application_name->bytes, (char *) (PIC(appnames_aliases[i].app_name)));
+            strlcpy((char *) application_name->bytes,
```

Comment:
Maybe we could check the return of strlcpy

Reply:
I'll rewrite this function, the original logic seems flawed.